File: pascal/dist.pas
program dist(input, output);

uses 
    Sysutils,
    Math;

function diferenca ( xp1:real; yp1:real; xp2:real; yp2:real):real;
var
    p1, p2: real;

begin
    p1 := Power((xp2 - xp1), 2);
    p2 := Power((yp2 - yp1), 2); 
    diferenca := sqrt( p1 + p2 );
end;

var
    xp1, yp1, xp2, yp2, resultado: real;

begin
    readLn(xp1, yp1);
    readLn(xp2, yp2);
    resultado := diferenca(xp1, yp1, xp2, yp2);
    writeln(Format('%.4f',[resultado])) ;
end.

File: cpp/dist.cpp
#include <iostream>
#include <math.h>
#include <iomanip>

using namespace std;

class Distancia{

    public:
        double xp1, xp2, yp1, yp2;

        double result( double xp1, double yp1, double xp2, double yp2 )
        {
            double res = ( (xp2 - xp1) * (xp2 - xp1)) + ((yp2 - yp1) * (yp2 - yp1));
            return sqrt(res);
        }

};

int main(void)
{
    Distancia dp;
    cin >> dp.xp1 >> dp.yp1;
    cin >> dp.xp2 >> dp.yp2;
    cout << fixed << setprecision(4) << dp.result( dp.xp1, dp.yp1, dp.xp2, dp.yp2) << "\n";
    return 0;
}